docs/reference/src/language/syntax/functions.md

```diff
@@ -1,15 +1,40 @@
 <!-- Copyright © SixtyFPS GmbH <info@slint.dev> ; SPDX-License-Identifier: MIT -->
 # Functions
 
-Declare helper functions with the `function` keyword.
-Functions are private by default, but can be made public with the `public` annotation.
+Functions in Slint are declared using the `function` keyword.
+
+Each function can have parameters which are declared within parentheses, following the format `name : type`.
+These parameters can be referenced by their names within the function body.
+
+Functions can also return a value. The return type is specified after `->` in the function signature.
+The `return` keyword is used within the function body to return an expression of the declared type.
+If a function does not explicitly return a value, the value of the last statement is returned by default.
+
+Functions can be annotated with the `pure` keyword.
+This indicates that the function does not cause any side effects.
+More details can be found in the [Purity](purity.md) chapter.
```

Comment:
Hm, could it be that this needs to be `../purity.md` since it's in `concepts/` while this file is in `syntax/`?

Reply:
You're right:

```suggestion
More details can be found in the [Purity](../concepts/purity.md) chapter.
```